app/questionnaire/questionnaire_schema.py

```diff
@@ -1352,26 +1380,34 @@ def _get_rules_section_dependencies(
             block_dependencies_for_progress_value_source,
         )
 
-    def _populate_calculated_summary_section_dependencies(self) -> None:
+    def _populate_calculation_summary_section_dependencies(self) -> None:
         """
-        For each block, find all the calculated summary value source dependencies
+        For each block, find all the calculated and grand calculated summary value source dependencies
         and make sure all involved sections are added as dependencies for that block and its section.
 
         Since calculated summaries can only contain answers from the section they are in,
-        it is sufficient to check the section for the calculated summary only.
+        it is sufficient to check the section for the calculated summary only. For grand calculated summaries
+        Need to include the section of the gcs, and each cs it references.
         """
         for section in self.get_sections():
             for block in self.get_blocks_for_section(section):
                 sources = get_mappings_with_key(
```

Comment:
I can't immediately see a nice way of making that work, it would need to check the mappings in the block to find out if it contains value sources, so I think it sort of has to check them all, but did you have a particular idea? If so maybe a follow up as its existing?